File: week07/15/5_Fri_search2D.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        row_num = len(matrix)
        row_col = len(matrix[0])

        r = row_num - 1
        c = 0
        while r >= 0 and c < row_col:
            if target == matrix[r][c]:
                return True
            elif target < matrix[r][c]:
                r -= 1
            else:
                c += 1
        
        return False

File: week07/15/5_Fri_search2D.cpp
#include <vector>

using namespace std;

class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int row_num = matrix.size();
        int col_num = matrix[0].size();

        int r = row_num - 1;
        int c = 0;

        while (r >= 0 && c < col_num) {
            if (target == matrix[r][c]) return true;
            else if (target < matrix[r][c]) r--;
            else c++;            
        } 
        return false;
    }
};